### ISSUE TITLE
Defaults showing in Theme Options when tampering with URL

### ISSUE THREAD

**Issue by parallelus** [parallelus/Runway-Framework#140]
It's possible to accidentally overwrite saved theme options with the defaults if you tamper with the URL after a save. This is specific to the theme options pages. To achieve this you need to do the following:

1. Edit and save the values of some theme options created with the Options Builder.
2. After saving, click on the browser URL field where the query string includes the save actions. Hit enter to reload the page with the current URL.
3. The reloaded page will show the default values for the options page instead of the ones from the database. Click "Save" again and it will overwrite the database values with defaults.

This can only happen if you do this odd page reloading method of clicking on the URL and hitting enter after a save, but the cause is how we are loading the values to the theme options based on the save action URL parameter. We aren't verifying a valid $_POST first so when no post variables exist it shows the defaults. When that happens and no post values exist but the save action is in the URL it should fall back to the database variables again.

**Comment by parallelus** [comment]
Is there a better way of doing this? The idea of reloading the page with JavaScript means it requires two page loads. It's not a question of performance but the delay and lack of efficiency bothers me.

I think the problem is deeper in the way we are validating the submission in the core admin object. If you look a the `request_handler()` function it checks the validation function but also sets the data again to the current submission even if the validation fails. This is likely to catch situations where a user has made many edits to a page but one specific validation fails. By reloading the posted values we save the user from having to re-enter the edits and they can fix the errors and resubmit. However, the process is failing because it's not making sure that a submission was posted.

We might want to look at how the `$this->extract_submission()` function handles these situations. It seems this alone might be able to ensure the proper performance. I didn't dig into this function much, but we might want to make the change here.

I've tested a change to the `request_handler()` by adding an additional check for `!empty($_POST)` before setting the data with `$this->extract_submission()` and it seems to work. 


Tested by changing [admin-object.php on line 515](https://github.com/parallelus/Runway-Framework/blob/master/runway-framework/framework/core/admin-object.php#L515) to: 
`if ( $this->action == 'save' && !empty($_POST) ) { `


There might be special cases where this doesn't work or other areas of the framework impacted. In my test of the theme options page it worked and also prevented errors. I was not able to get the default data to load through refreshing or editing the URL.
